### ISSUE TITLE
TODO: Protect datadog_at_fork_blocks from at_fork monkey patch with lock

### ISSUE THREAD

**Issue by ivoanjo** [DataDog/dd-trace-rb#3831]
As discussed in https://github.com/DataDog/dd-trace-rb/pull/3829#discussion_r1707144252 .

**Comment by ivoanjo** [comment]
cc @p-datadog 

I was going to pick this one up and add a mutex to the `AtForkMonkeyPatch, but actually I think it's no longer needed, especially after #3834 .

That is, in its current state we have:

```ruby
      module AtForkMonkeyPatch
        AT_FORK_CHILD_BLOCKS = [] # rubocop:disable Style/MutableConstant Used to store blocks to run, mutable by design.
		
        #...

        def self.run_at_fork_blocks(stage)
          raise(ArgumentError, "Unsupported stage #{stage}") unless stage == :child

          AT_FORK_CHILD_BLOCKS.each(&:call)
        end

        def self.at_fork(stage, &block)
          raise(ArgumentError, "Unsupported stage #{stage}") unless stage == :child
          raise(ArgumentError, 'Missing block argument') unless block

          AT_FORK_CHILD_BLOCKS << block

          true
        end
```

...and:
* Initialization is correct: it's done at require-time, and before requires are done there won't be any attempt to use the feature
 
* There can't be concurrency between `run_at_fork_blocks` and `at_fork`: By design, `run_at_fork_blocks` is called by our monkey patch just after fork, and "just after fork" there won't exist other threads running that can be accessing this

* Since we're the only users of this API, and we're not calling `at_fork` in background threads, there's also no concurrency between calls to `at_fork`.

Thus, while we could trivially add a `Mutex` here (I was half-way through doing it), I don't think we actually need one, given the current state of the code and how it's used.

I'm going to go ahead and close the issue, but ping me if you're not convinced by my hand-waving ;) :D
